### ISSUE TITLE
unable to use multi-dimensional arrays in TestCase initialization;  there is always compilation error CS0182	

### ISSUE THREAD

**Issue by petr12345** [nunit/nunit#4879]
In my testing project, I am using latest & greatest NUnit 4.2.2 and .NET 8.04.
Trying to use a multi-dimensional array in TestCase attribute, like in code the fragment below 

    [Test, Description("Tests metghod has expected behaviour")]
    [TestCase((uint)3, (uint)6, (uint)4, null, null)]
    [TestCase((uint)3, (uint)6, (uint)4, new int[] { 1 }, new int[, ] { { 1 } })]
    public void ApplyAllColumnsData_ExpectedBehavior(
        uint totalSections, 
        uint totalColumns, 
        uint totalRows,
        IEnumerable<int> incompleteSections,
        int[,] incompleteRowsIndexes)
    {
        // Arrange
  ...
  }

Regardless which variant of syntax I try to use in TestCase ( either new int[, ] { { 1 } })]  or new int[1, 1] { { 1 } })] or new int[][] { new int[] { 1 } }) or whatever else ), or what modification of last argument type I  try ( like IEnumerable<int[]>  incompleteRowsIndexes ),  there is always a compilation error

**Error (active)	CS0182	An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type ...**

In other words, this advice on StackOverdlow  https://stackoverflow.com/questions/29721444/using-the-testcase-attribute-with-a-two-dimensional-array  does not work for me.

I know one can resort to using TestCaseSource instead, but for consistency and readability reasons, I need to use TestCase.

Thanks for any help.

**Comment by jnm2** [comment]
There's nothing NUnit can do to make `[TestCase(...)]` contain a multi-dimensional array. This is a limitation of the C# language, as explained in the CS0182 compiler error. Consistency and readability are great goals, but they come second to language restrictions.

Besides TestCaseSource, you could also explore <https://docs.nunit.org/articles/nunit/writing-tests/attributes/valuesource.html> on the test method parameter itself. These are truly the only options possible if you're working with non-attribute-compatible types such as multidimensional arrays.
